### ISSUE TITLE
Why does .Net Core 2.1.1 Generic Host require the huge Microsoft.AspNetCore.App metapackage to work?

### ISSUE THREAD

**Issue by snielsson** [aspnet/Docs#7840]
I have asked on SO here : https://stackoverflow.com/questions/51553832/why-does-net-core-2-1-1-generic-host-require-the-huge-microsoft-aspnetcore-app

But let me repeat my question here:

I am trying to create a .Net Core 2.1.1 console app using the new [generic host](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-2.1).

It is a really simple program, just a .net core 2.1 console app looking like this:

    using System.Threading.Tasks;
    using Microsoft.Extensions.Hosting;
    
    namespace ConsoleApp1 {
    	internal class Program {
    		private async Task Main(string[] args)
    		{
    			await new HostBuilder()
    				.ConfigureAppConfiguration(x => x.SetBasePath("."))
    				.RunConsoleAsync();
    		}
    	}
    }

Instead of installing the huge package `Microsoft.AspNetCore.App` meta package I figured that I could just install the package `Microsoft.Extensions.Hosting v. 2.1.1`, so my `.csproj` becomes:

    <Project Sdk="Microsoft.NET.Sdk">
      <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>netcoreapp2.1</TargetFramework>
      </PropertyGroup>
      <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
        <LangVersion>latest</LangVersion>
      </PropertyGroup>
      <ItemGroup>
        <PackageReference Include="Microsoft.Extensions.Hosting" Version="2.1.1" />
      </ItemGroup>
    </Project>

This package installs a number of other packages:

 - Microsoft.Extensions.Configuration (2.1.1)
 - Microsoft.Extensions.DependencyInjection (2.1.1)
 - Microsoft.Extensions.FileProviders.Physical (2.1.1)
 - Microsoft.Extensions.Hosting.Abstractions (2.1.1)
 - Microsoft.Extensions.Logging (2.1.1)

When I do this I run into trouble because of the extension method `IConfigurationBulider.SetBasePath` (implemented in `Microsoft.Extensions.Configuration.FileConfigurationExtensions`) cannot be resolved/found by the compiler/build process.

I have tried fully qualifying the extension method, but it is as if the relevant assemblies from the packages are not referenced by the project.

The solution is to add the huge Microsoft.AspNetCore.App metapackage and then it works. But I find it strange that I need to add an AspNetCore package when my app has nothing to do with AspNet Core. 

1. Why is it neccessary to add Microsoft.AspNetCore.App and can it be avoided somehow ?
2. Is it just a temporary issue until the WebHostBuilder has been phased out in favor of the generic host builder ?

**Comment by guardrex** [comment]
> huge

Package trimming, although somewhat rudimentary according to `@DamianEdwards`, should pare it down. If you compare SxS the published output of your current app and an app that just references the `App` package, you shouldn't see much difference in size. As you suggest, just going with `App` is soooooooo much easier when it comes to building and maintaining an app.

Regarding the specific issue: Indeed, our Generic Host sample *does* go with the individual package approach. We should confer with @Tratcher on whether not it should go with `App`. (We may have discussed it when the topic was written, but I don't recall if we did or what was said.)

Just add a package reference for `Microsoft.Extensions.Configuration.FileExtensions` ...

```xml
<PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="2.1.1" />
```

We just added a bit to the top of the Configuration topic on this three days ago on #7771: https://docs.microsoft.com/aspnet/core/fundamentals/configuration ... it's near the top of that topic.
